libtenzir/src/type.cpp

```diff
@@ -61,7 +61,7 @@ constexpr size_t reserved_string_size(std::string_view str) {
   // table. It adds an extra byte because strings in FlatBuffers tables are
   // always zero-terminated, and then rounds up to a full four bytes because of
   // the included padding.
-  return str.empty() ? 0 : (((str.size() + 1 + 3) / 4) * 4);
+  return ((str.size() + 1 + 3) / 4) * 4;
```

Comment:
Can you add a small unit test for a record with an empty field name? That should crash in CI if this isn't correct.